src/components/pages/PassportChangesRedirect/index.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { routes } from 'src/constants/routes';
+import { Redirect } from 'react-router';
+import { createRouteUrl, IParsedQueryString } from 'src/utils/nav';
+import { parse } from 'query-string';
+
+export const PassportChangesRedirect = ({ location }) => {
+  const newLocation = { ...location };
```

Comment:
There is no need in modifying 'location' object. Modification `newLocation.pathname = ...` does not have any impact, because in `createRouteUrl` function `location` is only used for query string extraction - path is ignored

Reply:
Which means there’s no need to pass the whole location object ;) and maybe a 3rd argument as well...